src/java.desktop/windows/native/libawt/windows/WPrinterJob.cpp

```diff
@@ -133,12 +133,25 @@ static jobjectArray getPrinterNames(JNIEnv *env, DWORD flags) {
     jobjectArray nameArray;
 
     try {
-        ::EnumPrinters(flags,
-                       NULL, 4, NULL, 0, &cbNeeded, &cReturned);
-        pPrinterEnum = new BYTE[cbNeeded];
-        ::EnumPrinters(flags,
-                       NULL, 4, pPrinterEnum, cbNeeded, &cbNeeded,
-                       &cReturned);
+        ::EnumPrinters(flags, NULL, 4, NULL,
```

Comment:
Don't we need to check that this method call succeeds? Probably it is a crash reason when the EnumPrinters fails but we use cbNeeded anyway for array allocation?

Reply:
I guess since we are changing this method anyway, can we use PRINTER_ENUM_CONNECTIONS flag instead of hardcoded 4 so that it is more informative!!

Reply:
ok, it is for flags and not for level. please ignore

Reply:
> 
> 
> Don't we need to check that this method call succeeds? Probably it is a crash reason when the EnumPrinters fails but we use cbNeeded anyway for array allocation?

`EnumPrinters` always returns failure here because zero-sized buffer is too small to contain any data.